src/lib/dialog/dialog-ref.ts

```diff
@@ -93,6 +95,7 @@ export class MatDialogRef<T, R = any> {
       this._overlayRef.detachBackdrop();
     });
 
+    this._containerInstance._closeFocusOrigin = closeFocusOrigin;
```

Comment:
Rather than assigning the origin here as a property, you could pass it in to `_startExitAnimation`. It isn't a lot cleaner, but at least it keeps all the logic in the same class and you can turn the `_closeFocusOrigin` into a private property.

Reply:
Sounds good. I didn't go with that approach initially because I wasn't sure whether this technically belongs to an "animation".

Reply:
It doesn't, but it could be renamed to somthing like `_startExitSequence`.